tests/end2end/features/fileselect.feature

```diff
@@ -0,0 +1,36 @@
+# vim: ft=cucumber fileencoding=utf-8 sts=4 sw=4 et:
+
+Feature: Selecting files
+
+    Background:
+        Given I have a fresh instance
+
+    ## select single file
+
+    Scenario: Select one file with single command
+        When I set up a fake single file fileselector selecting "tests/end2end/data/numbers/1.txt"
+        And I open data/fileselect/fileselect.html
+        And I run :click-element id single_file
+        Then the javascript message "File selected: 1.txt" should be logged
+
+    Scenario: Select two files with single command
+        When I set up a fake single file fileselector selecting "tests/end2end/data/numbers/1.txt tests/end2end/data/numbers/2.txt"
+        And I open data/fileselect/fileselect.html
+        And I run :click-element id single_file
+        Then the javascript message "File selected: 1.txt" should be logged
+        Then the javascript message "File selected: 2.txt" should not be logged
```

Comment:
This should now be `Files: 1.txt, 2.txt` (same for the other places in here as well).

Reply:
True! Although in this scenario it should be only `Files: 1.txt` right? Since we return only the first entry if the `single_file.command` wrongly returns multiple files. I also fixed another small issue with the test file in this case

Reply:
Err, yes, I commented on the wrong line.

Reply:
I think things might be working now, let's see if the pipeline pass. Btw, does it declare failure in the case of a warning being raised? Since the above scenario will raise a warning. Got me thinking, could that warning be asserted perhaps?

Reply:
> Btw, does it declare failure in the case of a warning being raised? Since the above scenario will raise a warning. Got me thinking, could that warning be asserted perhaps?

I guess the failing tests answer that question :wink:

You can add `And the warning "..." should be shown` to the "Then" block to mark it as expected (and also assert it).